diff -Nru a/defines.h b/defines.h
--- a/defines.h	2013-03-07 10:06:13.000000000 +0100
+++ b/defines.h	2013-08-21 22:27:04.145035239 +0200
@@ -26,6 +26,7 @@
 #define _DEFINES_H
 
 /* $Id: defines.h,v 1.171 2013/03/07 09:06:13 dtucker Exp $ */
+#define MISSING_HOWMANY
 
 
 /* Constants */
diff -Nru a/loginrec.c b/loginrec.c
--- a/loginrec.c	2013-02-22 23:12:24.000000000 +0100
+++ b/loginrec.c	2013-08-21 22:43:45.868134274 +0200
@@ -180,6 +180,9 @@
 # include <util.h>
 #endif
 
+#include <sys/time.h>
+
+
 /**
  ** prototypes for helper functions in this file
  **/
@@ -780,19 +783,6 @@
 	if (li->hostaddr.sa.sa_family == AF_INET)
 		utx->ut_addr = li->hostaddr.sa_in.sin_addr.s_addr;
 # endif
-# ifdef HAVE_ADDR_V6_IN_UTMP
-	/* this is just a 128-bit IPv6 address */
-	if (li->hostaddr.sa.sa_family == AF_INET6) {
-		sa6 = ((struct sockaddr_in6 *)&li->hostaddr.sa);
-		memcpy(ut->ut_addr_v6, sa6->sin6_addr.s6_addr, 16);
-		if (IN6_IS_ADDR_V4MAPPED(&sa6->sin6_addr)) {
-			ut->ut_addr_v6[0] = ut->ut_addr_v6[3];
-			ut->ut_addr_v6[1] = 0;
-			ut->ut_addr_v6[2] = 0;
-			ut->ut_addr_v6[3] = 0;
-		}
-	}
-# endif
 # ifdef HAVE_SYSLEN_IN_UTMPX
 	/* ut_syslen is the length of the utx_host string */
 	utx->ut_syslen = MIN(strlen(li->hostname), sizeof(utx->ut_host));
diff -Nru a/sandbox-seccomp-filter.c b/sandbox-seccomp-filter.c
--- a/sandbox-seccomp-filter.c	2013-02-22 01:37:00.000000000 +0100
+++ b/sandbox-seccomp-filter.c	2013-08-21 22:45:30.634236354 +0200
@@ -44,6 +44,7 @@
 #include <linux/audit.h>
 #include <linux/filter.h>
 #include <linux/seccomp.h>
+#include <linux/prctl.h>
 #include <elf.h>
 
 #include <asm/unistd.h>
diff -Nru a/sshd.c b/sshd.c
--- a/sshd.c	2013-02-12 01:04:48.000000000 +0100
+++ b/sshd.c	2013-08-21 22:34:56.285508485 +0200
@@ -71,6 +71,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <sys/param.h>
 
 #include <openssl/dh.h>
 #include <openssl/bn.h>
diff -ru a/sshd_config b/sshd_config
--- a/sshd_config	2013-02-12 01:02:09.000000000 +0100
+++ b/sshd_config	2014-03-25 22:59:30.000000000 +0100
@@ -37,7 +37,7 @@
 # Authentication:
 
 #LoginGraceTime 2m
-#PermitRootLogin yes
+PermitRootLogin no
 #StrictModes yes
 #MaxAuthTries 6
 #MaxSessions 10
