diff -rup a/configure.ac b/configure.ac
--- a/configure.ac	2015-03-27 18:58:43.247905527 +0100
+++ b/configure.ac	2015-03-27 18:58:50.702923962 +0100
@@ -976,16 +976,22 @@ fi
 #                                     curl
 # ------------------------------------------------------------------------------
 
-AC_PATH_PROG(CURL_CONFIG, curl-config)
-if test "x$CURL_CONFIG" = "x" ; then
+AC_CHECK_HEADER(curl/curl.h,ac_found_libcurl_header="yes",ac_found_libcurl_header="no")
+AC_CHECK_LIB(curl,curl_version,ac_found_libcurl_lib="yes",ac_found_libcurl_lib="no")
+
+AC_MSG_CHECKING(for libcurl headers and librairy)
+if test "x$ac_found_libcurl_header" = "xno" -o "x$ac_found_libcurl_lib" = "xno" ; then
+    AC_MSG_RESULT(no)
     AC_MSG_ERROR([
-*** "curl-config" couldn't be found on your system.
-*** Try to install libcurl-dev with your software package manager.])
+*** zlib was not found. You may want to get it from http://zlib.net/
+*** or try to install libcurl-dev with your software package manager.])
+else
+    AC_MSG_RESULT(yes)
+    CURL_CFLAGS=`pkg-config libcurl --cflags`
+    CURL_LFLAGS=`pkg-config libcurl --libs`
+    AC_SUBST(CURL_CFLAGS)
+    AC_SUBST(CURL_LFLAGS)
 fi
-CURL_CFLAGS=`$CURL_CONFIG --cflags`
-CURL_LFLAGS=`$CURL_CONFIG --libs`
-AC_SUBST(CURL_CFLAGS)
-AC_SUBST(CURL_LFLAGS)
 
 # ------------------------------------------------------------------------------
 #                                    tests
