diff -Nru psmisc-22.20/doc/Makefile.in psmisc-22.20-new/doc/Makefile.in
--- psmisc-22.20/doc/Makefile.in	2012-09-19 14:29:26.000000000 +0200
+++ psmisc-22.20-new/doc/Makefile.in	2013-08-02 21:01:09.839993244 +0200
@@ -137,7 +137,7 @@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 INTLLIBS = @INTLLIBS@
 INTL_MACOSX_LIBS = @INTL_MACOSX_LIBS@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ @LTLIBINTL@
 LIBICONV = @LIBICONV@
 LIBINTL = @LIBINTL@
 LIBOBJS = @LIBOBJS@
diff -Nru psmisc-22.20/icons/Makefile.in psmisc-22.20-new/icons/Makefile.in
--- psmisc-22.20/icons/Makefile.in	2012-09-19 14:29:26.000000000 +0200
+++ psmisc-22.20-new/icons/Makefile.in	2013-08-02 21:01:09.839993244 +0200
@@ -104,7 +104,7 @@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 INTLLIBS = @INTLLIBS@
 INTL_MACOSX_LIBS = @INTL_MACOSX_LIBS@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ @LTLIBINTL@
 LIBICONV = @LIBICONV@
 LIBINTL = @LIBINTL@
 LIBOBJS = @LIBOBJS@
diff -Nru psmisc-22.20/Makefile.in psmisc-22.20-new/Makefile.in
--- psmisc-22.20/Makefile.in	2012-09-19 14:29:26.000000000 +0200
+++ psmisc-22.20-new/Makefile.in	2013-08-02 21:01:09.849993244 +0200
@@ -165,7 +165,7 @@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 INTLLIBS = @INTLLIBS@
 INTL_MACOSX_LIBS = @INTL_MACOSX_LIBS@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ @LTLIBINTL@
 LIBICONV = @LIBICONV@
 LIBINTL = @LIBINTL@
 LIBOBJS = @LIBOBJS@
diff -Nru psmisc-22.20/src/Makefile.in psmisc-22.20-new/src/Makefile.in
--- psmisc-22.20/src/Makefile.in	2012-09-19 14:29:26.000000000 +0200
+++ psmisc-22.20-new/src/Makefile.in	2013-08-02 21:01:09.849993244 +0200
@@ -158,7 +158,7 @@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 INTLLIBS = @INTLLIBS@
 INTL_MACOSX_LIBS = @INTL_MACOSX_LIBS@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ @LTLIBINTL@
 LIBICONV = @LIBICONV@
 LIBINTL = @LIBINTL@
 LIBOBJS = @LIBOBJS@
diff -Nru psmisc-22.20/src/pstree.c psmisc-22.20-new/src/pstree.c
--- psmisc-22.20/src/pstree.c	2012-09-19 15:05:43.000000000 +0200
+++ psmisc-22.20-new/src/pstree.c	2013-08-02 20:53:45.479996716 +0200
@@ -49,8 +49,6 @@
 #include <selinux/selinux.h>
 #endif                                /*WITH_SELINUX */
 
-extern const char *__progname;
-
 #define PROC_BASE    "/proc"
 
 #if defined(__FreeBSD_kernel__) || defined(__FreeBSD__)
@@ -912,7 +910,7 @@
     textdomain(PACKAGE);
 #endif
 
-    if (!strcmp(__progname, "pstree.x11"))
+    if (!strcmp(argv[0], "pstree.x11"))
         wait_end = 1;
 
     /*
diff -Nru psmisc-22.20/testsuite/Makefile.in psmisc-22.20-new/testsuite/Makefile.in
--- psmisc-22.20/testsuite/Makefile.in	2012-09-19 14:29:26.000000000 +0200
+++ psmisc-22.20-new/testsuite/Makefile.in	2013-08-02 21:01:09.859993244 +0200
@@ -108,7 +108,7 @@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 INTLLIBS = @INTLLIBS@
 INTL_MACOSX_LIBS = @INTL_MACOSX_LIBS@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ @LTLIBINTL@
 LIBICONV = @LIBICONV@
 LIBINTL = @LIBINTL@
 LIBOBJS = @LIBOBJS@
