diff -rup rsyslog-7.6.1-old/tools/omfile.c rsyslog-7.6.1/tools/omfile.c
--- rsyslog-7.6.1-old/tools/omfile.c	2015-07-31 11:25:02.286325154 +0200
+++ rsyslog-7.6.1/tools/omfile.c	2015-07-31 11:25:21.997384131 +0200
@@ -48,9 +48,7 @@
 #include <libgen.h>
 #include <unistd.h>
 #include <sys/file.h>
-#ifdef OS_SOLARIS
-#	include <fcntl.h>
-#endif
+#include <fcntl.h>
 #ifdef HAVE_ATOMIC_BUILTINS
 #	include <pthread.h>
 #endif
diff -rup rsyslog-7.6.1-old/tools/pidfile.c rsyslog-7.6.1/tools/pidfile.c
--- rsyslog-7.6.1-old/tools/pidfile.c	2015-07-31 11:31:49.496558569 +0200
+++ rsyslog-7.6.1/tools/pidfile.c	2015-07-31 11:32:55.454756929 +0200
@@ -36,9 +36,7 @@
 #include <string.h>
 #include <errno.h>
 #include <signal.h>
-#ifdef __sun
 #include <fcntl.h>
-#endif
 
 #include "srUtils.h"
 
diff -rup rsyslog-7.6.1-old/tools/syslogd.c rsyslog-7.6.1/tools/syslogd.c
--- rsyslog-7.6.1-old/tools/syslogd.c	2015-07-31 11:31:49.496558569 +0200
+++ rsyslog-7.6.1/tools/syslogd.c	2015-07-31 11:33:28.366855901 +0200
@@ -55,10 +55,10 @@
 #include <stdarg.h>
 #include <time.h>
 #include <assert.h>
+#include <fcntl.h>
 
 #ifdef OS_SOLARIS
 #	include <errno.h>
-#	include <fcntl.h>
 #	include <stropts.h>
 #	include <sys/termios.h>
 #	include <sys/types.h>
