diff -rup a/configure b/configure
--- a/configure	2014-05-09 18:49:42.000000000 +0200
+++ b/configure	2016-02-28 10:41:59.000000000 +0100
@@ -783,6 +783,7 @@ includedir
 localstatedir
 sharedstatedir
 sysconfdir
+sysroot
 datadir
 datarootdir
 libexecdir
@@ -882,6 +883,7 @@ libexecdir='${exec_prefix}/libexec'
 datarootdir='${prefix}/share'
 datadir='${datarootdir}'
 sysconfdir='${prefix}/etc'
+sysroot='/'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
 includedir='${prefix}/include'
@@ -1170,6 +1172,11 @@ do
   | --syscon=* | --sysco=* | --sysc=* | --sys=* | --sy=*)
     sysconfdir=$ac_optarg ;;
 
+  -sysroot | --sysroot)
+    ac_prev=sysroot ;;
+  -sysroot=* | --sysroot=*)
+    sysroot=$ac_optarg ;;
+
   -target | --target | --targe | --targ | --tar | --ta | --t)
     ac_prev=target_alias ;;
   -target=* | --target=* | --targe=* | --targ=* | --tar=* | --ta=* | --t=*)
@@ -1271,7 +1278,7 @@ fi
 
 # Check all directory arguments for consistency.
 for ac_var in	exec_prefix prefix bindir sbindir libexecdir datarootdir \
-		datadir sysconfdir sharedstatedir localstatedir includedir \
+		datadir sysconfdir sysroot sharedstatedir localstatedir includedir \
 		oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
 		libdir localedir mandir
 do
@@ -1424,6 +1431,7 @@ Fine tuning of the installation director
   --sbindir=DIR           system admin executables [EPREFIX/sbin]
   --libexecdir=DIR        program executables [EPREFIX/libexec]
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
+  --sysroot=DIR           sysroot []
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
   --libdir=DIR            object code libraries [EPREFIX/lib]
diff -rup a/lib/commonio.c b/lib/commonio.c
--- a/lib/commonio.c	2014-05-08 10:43:11.000000000 +0200
+++ b/lib/commonio.c	2016-02-28 10:47:39.000000000 +0100
@@ -466,8 +466,8 @@ static void dec_lock_count (void)
 			/* Tell nscd when lock count goes to zero,
 			   if any of the files were changed.  */
 			if (nscd_need_reload) {
-				nscd_flush_cache ("passwd");
-				nscd_flush_cache ("group");
+
+
 				nscd_need_reload = false;
 			}
 #ifdef HAVE_LCKPWDF
diff -rup a/lib/defines.h b/lib/defines.h
--- a/lib/defines.h	2014-03-01 19:59:51.000000000 +0100
+++ b/lib/defines.h	2016-02-28 10:41:59.000000000 +0100
@@ -299,20 +299,20 @@ char *strchr (), *strrchr (), *strtok ()
 	(strncpy((A), (B), sizeof(A) - 1), (A)[sizeof(A) - 1] = '\0')
 
 #ifndef PASSWD_FILE
-#define PASSWD_FILE "/etc/passwd"
+#define PASSWD_FILE SYSROOT"/etc/passwd"
 #endif
 
 #ifndef GROUP_FILE
-#define GROUP_FILE "/etc/group"
+#define GROUP_FILE SYSROOT"/etc/group"
 #endif
 
 #ifndef SHADOW_FILE
-#define SHADOW_FILE "/etc/shadow"
+#define SHADOW_FILE SYSROOT"/etc/shadow"
 #endif
 
 #ifdef SHADOWGRP
 #ifndef SGROUP_FILE
-#define SGROUP_FILE "/etc/gshadow"
+#define SGROUP_FILE SYSROOT"/etc/gshadow"
 #endif
 #endif
 
diff -rup a/lib/getdef.c b/lib/getdef.c
--- a/lib/getdef.c	2014-05-08 10:43:12.000000000 +0200
+++ b/lib/getdef.c	2016-02-28 10:41:59.000000000 +0100
@@ -139,7 +139,7 @@ static struct itemdef def_table[] = {
 };
 
 #ifndef LOGINDEFS
-#define LOGINDEFS "/etc/login.defs"
+#define LOGINDEFS SYSROOT"/etc/login.defs"
 #endif
 
 static char def_fname[] = LOGINDEFS;	/* login config defs file       */
diff -rup a/lib/gshadow_.h b/lib/gshadow_.h
--- a/lib/gshadow_.h	2014-03-01 19:59:51.000000000 +0100
+++ b/lib/gshadow_.h	2016-02-28 10:41:59.000000000 +0100
@@ -71,5 +71,5 @@ void endsgent ();
 int putsgent ();
 #endif
 
-#define	GSHADOW	"/etc/gshadow"
+#define	GSHADOW	SYSROOT"/etc/gshadow"
 #endif				/* ifndef _H_GSHADOW */
diff -rup a/lib/Makefile.am b/lib/Makefile.am
--- a/lib/Makefile.am	2014-03-01 19:59:51.000000000 +0100
+++ b/lib/Makefile.am	2016-02-28 10:41:59.000000000 +0100
@@ -2,6 +2,7 @@
 AUTOMAKE_OPTIONS = 1.0 foreign
 
 DEFS = 
+AM_CPPFLAGS = -DSYSROOT=\"$(sysroot)\"
 
 noinst_LTLIBRARIES = libshadow.la
 
diff -rup a/lib/Makefile.in b/lib/Makefile.in
--- a/lib/Makefile.in	2014-05-09 18:49:45.000000000 +0200
+++ b/lib/Makefile.in	2016-02-28 10:41:59.000000000 +0100
@@ -324,11 +324,13 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
 sysconfdir = @sysconfdir@
+sysroot = @sysroot@
 target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.0 foreign
+AM_CPPFLAGS = -DSYSROOT=\"$(sysroot)\"
 noinst_LTLIBRARIES = libshadow.la
 libshadow_la_LDFLAGS = -version-info 0:0:0
 libshadow_la_SOURCES = commonio.c commonio.h defines.h encrypt.c \
diff -rup a/lib/port.h b/lib/port.h
--- a/lib/port.h	2014-03-01 19:59:51.000000000 +0100
+++ b/lib/port.h	2016-02-28 10:41:59.000000000 +0100
@@ -52,7 +52,7 @@
  * PORT_DAY - Day of the week to a bit value (0 = Sunday).
  */
 
-#define	PORTS	"/etc/porttime"
+#define	PORTS	SYSROOT"/etc/porttime"
 #define	PORT_IDS	64
 #define	PORT_TTY	64
 #define	PORT_TIMES	24
diff -rup a/libmisc/limits.c b/libmisc/limits.c
--- a/libmisc/limits.c	2014-03-01 19:59:51.000000000 +0100
+++ b/libmisc/limits.c	2016-02-28 10:41:59.000000000 +0100
@@ -56,7 +56,7 @@
 #endif
 #ifdef LIMITS
 #ifndef LIMITS_FILE
-#define LIMITS_FILE "/etc/limits"
+#define LIMITS_FILE SYSROOT"/etc/limits"
 #endif
 #define LOGIN_ERROR_RLIMIT	1
 #define LOGIN_ERROR_LOGIN	2
diff -rup a/libmisc/Makefile.am b/libmisc/Makefile.am
--- a/libmisc/Makefile.am	2014-03-01 19:59:51.000000000 +0100
+++ b/libmisc/Makefile.am	2016-02-28 10:41:59.000000000 +0100
@@ -1,6 +1,8 @@
 
 EXTRA_DIST = .indent.pro xgetXXbyYY.c
 
+AM_CPPFLAGS = -DSYSROOT=\"$(sysroot)\"
+
 INCLUDES = -I$(top_srcdir)/lib
 
 noinst_LIBRARIES = libmisc.a
diff -rup a/libmisc/Makefile.in b/libmisc/Makefile.in
--- a/libmisc/Makefile.in	2014-05-09 18:49:45.000000000 +0200
+++ b/libmisc/Makefile.in	2016-02-28 10:41:59.000000000 +0100
@@ -343,11 +343,13 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
 sysconfdir = @sysconfdir@
+sysroot = @sysroot@
 target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 EXTRA_DIST = .indent.pro xgetXXbyYY.c
+AM_CPPFLAGS = -DSYSROOT=\"$(sysroot)\"
 INCLUDES = -I$(top_srcdir)/lib
 noinst_LIBRARIES = libmisc.a
 libmisc_a_SOURCES = \
diff -rup a/libmisc/xgetpwnam.c b/libmisc/xgetpwnam.c
--- a/libmisc/xgetpwnam.c	2014-03-01 18:50:05.000000000 +0100
+++ b/libmisc/xgetpwnam.c	2016-02-28 10:41:59.000000000 +0100
@@ -49,7 +49,12 @@
  * This file provide wrapper to the getpwnam or getpwnam_r functions.
  */
 
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
 #include <config.h>
+#include <pthread.h>
+#include <errno.h>
 
 #include "pwio.h"
 
@@ -60,5 +65,106 @@
 #define DUP_FUNCTION	__pw_dup
 #define HAVE_FUNCTION_R (defined HAVE_GETPWNAM_R)
 
+static FILE *f;
+
+static unsigned atou(char **s)
+{
+    unsigned x;
+    for (x=0; **s-'0'<10U; ++*s) x=10*x+(**s-'0');
+    return x;
+}
+
+struct passwd *sysroot_getpwent_a(FILE *f, struct passwd *pw, char **line,
+        size_t *size)
+{
+    ssize_t l;
+    char *s;
+    int cs;
+    pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
+    for (;;) {
+        if ((l=getline(line, size, f)) < 0) {
+            free(*line);
+            *line = 0;
+            pw = 0;
+            break;
+        }
+        line[0][l-1] = 0;
+
+        s = line[0];
+        pw->pw_name = s++;
+        if (!(s = strchr(s, ':'))) continue;
+
+        *s++ = 0; pw->pw_passwd = s;
+        if (!(s = strchr(s, ':'))) continue;
+
+        *s++ = 0; pw->pw_uid = atou(&s);
+        if (*s != ':') continue;
+
+        *s++ = 0; pw->pw_gid = atou(&s);
+        if (*s != ':') continue;
+
+        *s++ = 0; pw->pw_gecos = s;
+        if (!(s = strchr(s, ':'))) continue;
+
+        *s++ = 0; pw->pw_dir = s;
+        if (!(s = strchr(s, ':'))) continue;
+
+        *s++ = 0; pw->pw_shell = s;
+        break;
+    }
+    pthread_setcancelstate(cs, 0);
+    return pw;
+}
+
+
+#define FIX(x) (pw->pw_##x = pw->pw_##x-line+buf)
+
+static int sysroot_getpw_r(const char *name, uid_t uid, struct passwd *pw, char *buf, size_t size, struct passwd **res)
+{
+    FILE *f;
+    char *line = 0;
+    size_t len = 0;
+    int rv = 0;
+    int cs;
+
+    pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
+
+    f = fopen(SYSROOT"/etc/passwd", "rbe");
+    if (!f) {
+        rv = errno;
+        goto done;
+    }
+
+    *res = 0;
+    while (sysroot_getpwent_a(f, pw, &line, &len)) {
+        if (name && !strcmp(name, pw->pw_name)
+        || !name && pw->pw_uid == uid) {
+            if (size < len) {
+                rv = ERANGE;
+                break;
+            }
+            *res = pw;
+            memcpy(buf, line, len);
+            FIX(name);
+            FIX(passwd);
+            FIX(gecos);
+            FIX(dir);
+            FIX(shell);
+            break;
+        }
+    }
+    free(line);
+    fclose(f);
+done:
+    pthread_setcancelstate(cs, 0);
+    return rv;
+}
+
+int getpwnam_r(const char *name, struct passwd *pw, char *buf, size_t size, struct passwd **res)
+{
+    return sysroot_getpw_r(name, 0, pw, buf, size, res);
+}
+
+
 #include "xgetXXbyYY.c"
 
diff -rup a/src/chage.c b/src/chage.c
--- a/src/chage.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/chage.c	2016-02-28 10:41:59.000000000 +0100
@@ -816,7 +816,7 @@ int main (int argc, char **argv)
 
 	ruid = getuid ();
 	rgid = getgid ();
-	amroot = (ruid == 0);
+	amroot = 1; /* (ruid == 0); */
 #ifdef WITH_SELINUX
 	if (amroot && (is_selinux_enabled () > 0)) {
 		amroot = (selinux_check_passwd_access (PASSWD__ROOTOK) == 0);
diff -rup a/src/chfn.c b/src/chfn.c
--- a/src/chfn.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/chfn.c	2016-02-28 10:47:39.000000000 +0100
@@ -745,7 +745,7 @@ int main (int argc, char **argv)
 
 	SYSLOG ((LOG_INFO, "changed user '%s' information", user));
 
-	nscd_flush_cache ("passwd");
+
 
 	closelog ();
 	exit (E_SUCCESS);
diff -rup a/src/chgpasswd.c b/src/chgpasswd.c
--- a/src/chgpasswd.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/chgpasswd.c	2016-02-28 10:47:39.000000000 +0100
@@ -580,7 +580,7 @@ int main (int argc, char **argv)
 
 	close_files ();
 
-	nscd_flush_cache ("group");
+
 
 	return (0);
 }
diff -rup a/src/chpasswd.c b/src/chpasswd.c
--- a/src/chpasswd.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/chpasswd.c	2016-02-28 10:47:39.000000000 +0100
@@ -623,7 +623,7 @@ int main (int argc, char **argv)
 		close_files ();
 	}
 
-	nscd_flush_cache ("passwd");
+
 
 	return (0);
 }
diff -rup a/src/chsh.c b/src/chsh.c
--- a/src/chsh.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/chsh.c	2016-02-28 10:47:39.000000000 +0100
@@ -56,7 +56,7 @@
 #include "exitcodes.h"
 
 #ifndef SHELLS_FILE
-#define SHELLS_FILE "/etc/shells"
+#define SHELLS_FILE SYSROOT"/etc/shells"
 #endif
 /*
  * Global variables
@@ -455,7 +455,7 @@ int main (int argc, char **argv)
 	/*
 	 * This command behaves different for root and non-root users.
 	 */
-	amroot = (getuid () == 0);
+	amroot = 1; //(getuid () == 0);
 
 	OPENLOG ("chsh");
 
@@ -556,7 +556,7 @@ int main (int argc, char **argv)
 
 	SYSLOG ((LOG_INFO, "changed user '%s' shell to '%s'", user, loginsh));
 
-	nscd_flush_cache ("passwd");
+
 
 	closelog ();
 	exit (E_SUCCESS);
diff -rup a/src/gpasswd.c b/src/gpasswd.c
--- a/src/gpasswd.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/gpasswd.c	2016-02-28 10:47:39.000000000 +0100
@@ -86,7 +86,7 @@ static char *myname = NULL;
 /* The UID of the caller */
 static uid_t bywho;
 /* Indicate if gpasswd was called by root */
-#define amroot	(0 == bywho)
+#define amroot	1 /* (0 == bywho) */
 
 /* The number of retries for th user to provide and repeat a new password */
 #ifndef	RETRIES
@@ -1183,12 +1183,14 @@ int main (int argc, char **argv)
 	 * output, etc.
 	 */
       output:
+/*
 	if (setuid (0) != 0) {
 		fputs (_("Cannot change ID to root.\n"), stderr);
 		SYSLOG ((LOG_ERR, "can't setuid(0)"));
 		closelog ();
 		exit (E_NOPERM);
 	}
+*/
 	pwd_init ();
 
 	open_files ();
@@ -1201,7 +1203,7 @@ int main (int argc, char **argv)
 
 	close_files ();
 
-	nscd_flush_cache ("group");
+
 
 	exit (E_SUCCESS);
 }
diff -rup a/src/groupadd.c b/src/groupadd.c
--- a/src/groupadd.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/groupadd.c	2016-02-28 10:47:39.000000000 +0100
@@ -617,7 +617,7 @@ int main (int argc, char **argv)
 	grp_update ();
 	close_files ();
 
-	nscd_flush_cache ("group");
+
 
 	return E_SUCCESS;
 }
diff -rup a/src/groupdel.c b/src/groupdel.c
--- a/src/groupdel.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/groupdel.c	2016-02-28 10:47:39.000000000 +0100
@@ -477,7 +477,7 @@ int main (int argc, char **argv)
 
 	close_files ();
 
-	nscd_flush_cache ("group");
+
 
 	return E_SUCCESS;
 }
diff -rup a/src/groupmod.c b/src/groupmod.c
--- a/src/groupmod.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/groupmod.c	2016-02-28 10:47:39.000000000 +0100
@@ -864,7 +864,7 @@ int main (int argc, char **argv)
 
 	close_files ();
 
-	nscd_flush_cache ("group");
+
 
 	return E_SUCCESS;
 }
diff -rup a/src/grpck.c b/src/grpck.c
--- a/src/grpck.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/grpck.c	2016-02-28 10:47:39.000000000 +0100
@@ -869,7 +869,7 @@ int main (int argc, char **argv)
 	/* Commit the change in the database if needed */
 	close_files (changed);
 
-	nscd_flush_cache ("group");
+
 
 	/*
 	 * Tell the user what we did and exit.
diff -rup a/src/grpconv.c b/src/grpconv.c
--- a/src/grpconv.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/grpconv.c	2016-02-28 10:47:39.000000000 +0100
@@ -271,7 +271,7 @@ int main (int argc, char **argv)
 		/* continue */
 	}
 
-	nscd_flush_cache ("group");
+
 
 	return 0;
 }
diff -rup a/src/grpunconv.c b/src/grpunconv.c
--- a/src/grpunconv.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/grpunconv.c	2016-02-28 10:47:39.000000000 +0100
@@ -235,7 +235,7 @@ int main (int argc, char **argv)
 		/* continue */
 	}
 
-	nscd_flush_cache ("group");
+
 
 	return 0;
 }
diff -rup a/src/login_nopam.c b/src/login_nopam.c
--- a/src/login_nopam.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/login_nopam.c	2016-02-28 10:41:59.000000000 +0100
@@ -59,7 +59,7 @@
 
  /* Path name of the access control file. */
 #ifndef	TABLE
-#define TABLE	"/etc/login.access"
+#define TABLE	SYSROOT"/etc/login.access"
 #endif
 
 /* Delimiters for fields and for lists of users, ttys or hosts. */
diff -rup a/src/logoutd.c b/src/logoutd.c
--- a/src/logoutd.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/logoutd.c	2016-02-28 10:41:59.000000000 +0100
@@ -50,7 +50,7 @@ const char *Prog;
 #endif
 
 #ifndef HUP_MESG_FILE
-#define HUP_MESG_FILE "/etc/logoutd.mesg"
+#define HUP_MESG_FILE SYSROOT"/etc/logoutd.mesg"
 #endif
 
 /* local function prototypes */
diff -rup a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2014-05-08 10:43:11.000000000 +0200
+++ b/src/Makefile.am	2016-02-28 10:41:59.000000000 +0100
@@ -3,7 +3,7 @@ EXTRA_DIST = \
 	.indent.pro
 
 ubindir = ${prefix}/bin
-usbindir = ${prefix}/sbin
+usbindir = ${prefix}/bin
 suidperms = 4755
 sgidperms = 2755
 
@@ -66,7 +66,7 @@ LDADD          = $(INTLLIBS) \
 		 $(LIBTCB) \
 		 $(top_builddir)/libmisc/libmisc.a \
 		 $(top_builddir)/lib/libshadow.la
-AM_CPPFLAGS    = -DLOCALEDIR=\"$(datadir)/locale\"
+AM_CPPFLAGS    = -DSYSROOT=\"$(sysroot)\" -DLOCALEDIR=\"$(datadir)/locale\" -DSG=\"`echo sg | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`\" -DVIGR=\"`echo vigr | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`\"
 
 if ACCT_TOOLS_SETUID
 LIBPAM_SUID  = $(LIBPAM)
@@ -118,13 +118,13 @@ vipw_LDADD     = $(LDADD) $(LIBSELINUX)
 
 install-am: all-am
 	$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
-	ln -sf newgrp	$(DESTDIR)$(ubindir)/sg
-	ln -sf vipw	$(DESTDIR)$(usbindir)/vigr
+	ln -sf `echo newgrp | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'` $(DESTDIR)$(ubindir)/`echo sg | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`
+	ln -sf `echo vipw | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'` $(DESTDIR)$(usbindir)/`echo vigr | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`
 	for i in $(suidbins); do \
-		chmod -f $(suidperms) $(DESTDIR)$(bindir)/$$i; \
+		chmod -f 4755 $(DESTDIR)$(bindir)/`echo $$i | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`; \
 	done
 	for i in $(suidubins); do \
-		chmod -f $(suidperms) $(DESTDIR)$(ubindir)/$$i; \
+		chmod -f 4755 $(DESTDIR)$(ubindir)/`echo $$i | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`; \
 	done
 if WITH_TCB
 	for i in $(shadowsgidubins); do \
diff -rup a/src/Makefile.in b/src/Makefile.in
--- a/src/Makefile.in	2014-05-09 18:49:48.000000000 +0200
+++ b/src/Makefile.in	2016-02-28 10:41:59.000000000 +0100
@@ -489,6 +489,7 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
 sysconfdir = @sysconfdir@
+sysroot = @sysroot@
 target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
@@ -497,7 +498,7 @@ EXTRA_DIST = \
 	.indent.pro
 
 ubindir = ${prefix}/bin
-usbindir = ${prefix}/sbin
+usbindir = ${prefix}/bin
 suidperms = 4755
 sgidperms = 2755
 INCLUDES = \
@@ -512,7 +513,7 @@ LDADD = $(INTLLIBS) \
 		 $(top_builddir)/libmisc/libmisc.a \
 		 $(top_builddir)/lib/libshadow.la
 
-AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\"
+AM_CPPFLAGS = -DSYSROOT=\"$(sysroot)\" -DLOCALEDIR=\"$(datadir)/locale\" -DSG=\"`echo sg | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`\" -DVIGR=\"`echo vigr | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`\"
 @ACCT_TOOLS_SETUID_FALSE@LIBPAM_SUID = 
 @ACCT_TOOLS_SETUID_TRUE@LIBPAM_SUID = $(LIBPAM)
 @USE_PAM_FALSE@LIBCRYPT_NOPAM = $(LIBCRYPT)
@@ -1221,13 +1222,13 @@ uninstall-am: uninstall-binPROGRAMS unin
 
 install-am: all-am
 	$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
-	ln -sf newgrp	$(DESTDIR)$(ubindir)/sg
-	ln -sf vipw	$(DESTDIR)$(usbindir)/vigr
+	ln -sf `echo newgrp | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'` $(DESTDIR)$(ubindir)/`echo sg | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`
+	ln -sf `echo vipw | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'` $(DESTDIR)$(usbindir)/`echo vigr | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`
 	for i in $(suidbins); do \
-		chmod -f $(suidperms) $(DESTDIR)$(bindir)/$$i; \
+		chmod -f 4755 $(DESTDIR)$(bindir)/`echo $$i | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`; \
 	done
 	for i in $(suidubins); do \
-		chmod -f $(suidperms) $(DESTDIR)$(ubindir)/$$i; \
+		chmod -f 4755 $(DESTDIR)$(ubindir)/`echo $$i | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`; \
 	done
 @WITH_TCB_TRUE@	for i in $(shadowsgidubins); do \
 @WITH_TCB_TRUE@		chown root:shadow $(DESTDIR)$(ubindir)/$$i; \
diff -rup a/src/newgrp.c b/src/newgrp.c
--- a/src/newgrp.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/newgrp.c	2016-02-28 10:41:59.000000000 +0100
@@ -306,7 +306,7 @@ static void syslog_sg (const char *name,
 		if ((pid_t)-1 == child) {
 			/* error in fork() */
 			fprintf (stderr, _("%s: failure forking: %s\n"),
-				 is_newgrp ? "newgrp" : "sg", strerror (errno));
+				 is_newgrp ? "newgrp" : SG, strerror (errno));
 #ifdef WITH_AUDIT
 			if (group) {
 				snprintf (audit_buf, sizeof(audit_buf),
@@ -429,7 +429,7 @@ int main (int argc, char **argv)
 	 */
 	Prog = Basename (argv[0]);
 	is_newgrp = (strcmp (Prog, "newgrp") == 0);
-	OPENLOG (is_newgrp ? "newgrp" : "sg");
+	OPENLOG (is_newgrp ? "newgrp" : SG);
 	gid = getgid ();
 	argc--;
 	argv++;
diff -rup a/src/newusers.c b/src/newusers.c
--- a/src/newusers.c	2014-05-09 12:20:28.000000000 +0200
+++ b/src/newusers.c	2016-02-28 10:47:39.000000000 +0100
@@ -1230,8 +1230,8 @@ int main (int argc, char **argv)
 
 	close_files ();
 
-	nscd_flush_cache ("passwd");
-	nscd_flush_cache ("group");
+
+
 
 #ifdef USE_PAM
 	unsigned int i;
diff -rup a/src/passwd.c b/src/passwd.c
--- a/src/passwd.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/passwd.c	2016-02-28 10:47:39.000000000 +0100
@@ -815,7 +815,7 @@ int main (int argc, char **argv)
 	 * The program behaves differently when executed by root than when
 	 * executed by a normal user.
 	 */
-	amroot = (getuid () == 0);
+	amroot = 1; //(getuid () == 0);
 
 	OPENLOG ("passwd");
 
@@ -1136,20 +1136,14 @@ int main (int argc, char **argv)
 		exit (E_SUCCESS);
 	}
 #endif				/* USE_PAM */
-	if (setuid (0) != 0) {
-		(void) fputs (_("Cannot change ID to root.\n"), stderr);
-		SYSLOG ((LOG_ERR, "can't setuid(0)"));
-		closelog ();
-		exit (E_NOPERM);
-	}
 	if (spw_file_present ()) {
 		update_shadow ();
 	} else {
 		update_noshadow ();
 	}
 
-	nscd_flush_cache ("passwd");
-	nscd_flush_cache ("group");
+
+
 
 	SYSLOG ((LOG_INFO, "password for '%s' changed by '%s'", name, myname));
 	closelog ();
diff -rup a/src/pwck.c b/src/pwck.c
--- a/src/pwck.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/pwck.c	2016-02-28 10:47:39.000000000 +0100
@@ -876,7 +876,7 @@ int main (int argc, char **argv)
 
 	close_files (changed);
 
-	nscd_flush_cache ("passwd");
+
 
 	/*
 	 * Tell the user what we did and exit.
diff -rup a/src/pwconv.c b/src/pwconv.c
--- a/src/pwconv.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/pwconv.c	2016-02-28 10:47:39.000000000 +0100
@@ -325,7 +325,7 @@ int main (int argc, char **argv)
 		/* continue */
 	}
 
-	nscd_flush_cache ("passwd");
+
 
 	return E_SUCCESS;
 }
diff -rup a/src/pwunconv.c b/src/pwunconv.c
--- a/src/pwunconv.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/pwunconv.c	2016-02-28 10:47:39.000000000 +0100
@@ -249,7 +249,7 @@ int main (int argc, char **argv)
 		/* continue */
 	}
 
-	nscd_flush_cache ("passwd");
+
 
 	return 0;
 }
diff -rup a/src/suauth.c b/src/suauth.c
--- a/src/suauth.c	2014-03-01 18:50:05.000000000 +0100
+++ b/src/suauth.c	2016-02-28 10:41:59.000000000 +0100
@@ -40,7 +40,7 @@
 #include "prototypes.h"
 
 #ifndef SUAUTHFILE
-#define SUAUTHFILE "/etc/suauth"
+#define SUAUTHFILE SYSROOT"/etc/suauth"
 #endif
 
 #define	NOACTION	0
diff -rup a/src/useradd.c b/src/useradd.c
--- a/src/useradd.c	2014-05-09 12:20:28.000000000 +0200
+++ b/src/useradd.c	2016-02-28 10:47:39.000000000 +0100
@@ -73,11 +73,11 @@
 #endif
 
 #ifndef SKEL_DIR
-#define SKEL_DIR "/etc/skel"
+#define SKEL_DIR SYSROOT"/etc/skel"
 #endif
 #ifndef USER_DEFAULTS_FILE
-#define USER_DEFAULTS_FILE "/etc/default/useradd"
-#define NEW_USER_FILE "/etc/default/nuaddXXXXXX"
+#define USER_DEFAULTS_FILE SYSROOT"/etc/default/useradd"
+#define NEW_USER_FILE SYSROOT"/etc/default/nuaddXXXXXX"
 #endif
 /*
  * Needed for MkLinux DR1/2/2.1 - J.
@@ -111,6 +111,7 @@ static uid_t user_id;
 static gid_t user_gid;
 static const char *user_comment = "";
 static const char *user_home = "";
+static int sysroot = 0;
 static const char *user_shell = "";
 static const char *create_mail_spool = "";
 #ifdef WITH_SELINUX
@@ -802,6 +803,8 @@ static void new_pwent (struct passwd *pw
 	pwent->pw_gid = user_gid;
 	pwent->pw_gecos = (char *) user_comment;
 	pwent->pw_dir = (char *) user_home;
+	if(sysroot)
+		pwent->pw_dir += strlen(SYSROOT);
 	pwent->pw_shell = (char *) user_shell;
 }
 
@@ -1348,14 +1351,15 @@ static void process_flags (int argc, cha
 		}
 		if (!dflg) {
 			char *uh;
-			size_t len = strlen (def_home) + strlen (user_name) + 2;
+			size_t len = strlen(SYSROOT) + strlen (def_home) + strlen (user_name) + 2;
 			int wlen;
 
 			uh = xmalloc (len);
-			wlen = snprintf (uh, len, "%s/%s", def_home, user_name);
+			wlen = snprintf (uh, len, SYSROOT "%s/%s", def_home, user_name);
 			assert (wlen == (int) len -1);
 
 			user_home = uh;
+			sysroot = 1;
 		}
 	}
 
@@ -1814,7 +1818,7 @@ static void usr_update (void)
 	SYSLOG ((LOG_INFO,
 	         "new user: name=%s, UID=%u, GID=%u, home=%s, shell=%s",
 	         user_name, (unsigned int) user_id,
-	         (unsigned int) user_gid, user_home, user_shell));
+	         (unsigned int) user_gid, user_home + (sysroot ? strlen(SYSROOT) : 0), user_shell));
 
 	/*
 	 * Initialize faillog and lastlog entries for this UID in case
@@ -1952,8 +1956,8 @@ static void create_mail (void)
 		if (NULL == spool) {
 			spool = "/var/mail";
 		}
-		file = alloca (strlen (spool) + strlen (user_name) + 2);
-		sprintf (file, "%s/%s", spool, user_name);
+		file = alloca (strlen(SYSROOT) + strlen (spool) + strlen (user_name) + 2);
+		sprintf (file, SYSROOT"%s/%s", spool, user_name);
 		fd = open (file, O_CREAT | O_WRONLY | O_TRUNC | O_EXCL, 0);
 		if (fd < 0) {
 			perror (_("Creating mailbox file"));
@@ -2088,7 +2092,7 @@ int main (int argc, char **argv)
 	/*
 	 * Start with a quick check to see if the user exists.
 	 */
-	if (getpwnam (user_name) != NULL) { /* local, no need for xgetpwnam */
+	if (xgetpwnam (user_name) != NULL) { /* local, no need for xgetpwnam */
 		fprintf (stderr, _("%s: user '%s' already exists\n"), Prog, user_name);
 #ifdef WITH_AUDIT
 		audit_logger (AUDIT_ADD_USER, Prog,
@@ -2239,8 +2243,8 @@ int main (int argc, char **argv)
 	}
 #endif				/* WITH_SELINUX */
 
-	nscd_flush_cache ("passwd");
-	nscd_flush_cache ("group");
+
+
 
 	return E_SUCCESS;
 }
diff -rup a/src/userdel.c b/src/userdel.c
--- a/src/userdel.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/userdel.c	2016-02-28 10:47:39.000000000 +0100
@@ -827,7 +827,8 @@ static int remove_mailbox (void)
 	if (NULL == maildir) {
 		return 0;
 	}
-	snprintf (mailfile, sizeof mailfile, "%s/%s", maildir, user_name);
+	snprintf (mailfile, sizeof mailfile, SYSROOT "%s/%s",
+			maildir, user_name);
 
 	if (access (mailfile, F_OK) != 0) {
 		if (ENOENT == errno) {
@@ -1275,8 +1276,8 @@ int main (int argc, char **argv)
 	errors += remove_tcbdir (user_name, user_id);
 #endif				/* WITH_TCB */
 
-	nscd_flush_cache ("passwd");
-	nscd_flush_cache ("group");
+
+
 
 	return ((0 != errors) ? E_HOMEDIR : E_SUCCESS);
 }
diff -rup a/src/usermod.c b/src/usermod.c
--- a/src/usermod.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/usermod.c	2016-02-28 10:47:39.000000000 +0100
@@ -2013,7 +2013,7 @@ static void move_mailbox (void)
 	 * replacing /var/spool/mail/luser with a hard link to /etc/passwd
 	 * between stat and chown).  --marekm
 	 */
-	(void) snprintf (mailfile, sizeof mailfile, "%s/%s",
+	(void) snprintf (mailfile, sizeof mailfile, SYSROOT"%s/%s",
 	                 maildir, user_name);
 	mailfile[(sizeof mailfile) - 1] = '\0';
 	fd = open (mailfile, O_RDONLY | O_NONBLOCK, 0);
@@ -2052,8 +2052,8 @@ static void move_mailbox (void)
 	(void) close (fd);
 
 	if (lflg) {
-		(void) snprintf (newmailfile, sizeof newmailfile, "%s/%s",
-		                 maildir, user_newname);
+		(void) snprintf (newmailfile, sizeof newmailfile,
+				SYSROOT "%s/%s", maildir, user_newname);
 		newmailfile[(sizeof newmailfile) - 1] = '\0';
 		if (   (link (mailfile, newmailfile) != 0)
 		    || (unlink (mailfile) != 0)) {
@@ -2193,8 +2193,8 @@ int main (int argc, char **argv)
 	}
 #endif
 
-	nscd_flush_cache ("passwd");
-	nscd_flush_cache ("group");
+
+
 
 #ifdef WITH_SELINUX
 	if (Zflg) {
diff -rup a/src/vipw.c b/src/vipw.c
--- a/src/vipw.c	2014-03-01 19:59:51.000000000 +0100
+++ b/src/vipw.c	2016-02-28 10:47:39.000000000 +0100
@@ -444,7 +444,7 @@ int main (int argc, char **argv)
 
 	process_root_flag ("-R", argc, argv);
 
-	do_vipw = (strcmp (Prog, "vigr") != 0);
+	do_vipw = (strcmp (Prog, VIGR) != 0);
 
 	OPENLOG (do_vipw ? "vipw" : "vigr");
 
@@ -554,8 +554,8 @@ int main (int argc, char **argv)
 #endif				/* SHADOWGRP */
 	}
 
-	nscd_flush_cache ("passwd");
-	nscd_flush_cache ("group");
+
+
 
 	return E_SUCCESS;
 }
