diff -rup libtool-2.4.2-old/libtoolize.in libtool-2.4.2/libtoolize.in
--- libtool-2.4.2-old/libtoolize.in	2015-03-29 19:51:21.754760126 +0200
+++ libtool-2.4.2/libtoolize.in	2015-03-29 19:56:34.353715177 +0200
@@ -2486,11 +2486,12 @@ func_check_macros ()
   all_pkgltdl_files="COPYING.LIB Makefile Makefile.in Makefile.inc Makefile.am README acinclude.m4 aclocal.m4 argz_.h argz.c config.h.in config-h.in configure configure.ac configure.in libltdl/lt__alloc.h libltdl/lt__dirent.h libltdl/lt__glibc.h libltdl/lt__private.h libltdl/lt__strl.h libltdl/lt_dlloader.h libltdl/lt_error.h libltdl/lt_system.h libltdl/slist.h loaders/dld_link.c loaders/dlopen.c loaders/dyld.c loaders/load_add_on.c loaders/loadlibrary.c loaders/preopen.c loaders/shl_load.c lt__alloc.c lt__dirent.c lt__strl.c lt_dlloader.c lt_error.c ltdl.c ltdl.h slist.c"
 
   # Locations for important files:
-  prefix=@prefix@
-  datadir=@datadir@
-  pkgdatadir=@pkgdatadir@
-  pkgltdldir=@pkgdatadir@
-  aclocaldir=@aclocaldir@
+  progdir=$(dirname $progpath)
+  prefix=$progdir/..
+  datadir=$prefix/@datadir@
+  pkgdatadir=$prefix/@pkgdatadir@
+  pkgltdldir=$prefix/@pkgdatadir@
+  aclocaldir=$prefix/@aclocaldir@
   auxdir=
   macrodir=
   configure_ac=configure.in
