diff -rup a/evdev.c b/evdev.c
--- a/evdev.c	2015-07-28 11:33:51.791462637 +0200
+++ b/evdev.c	2015-07-28 11:35:29.456826963 +0200
@@ -30,6 +30,7 @@
 
 #ifdef HAVE_LINUX_INPUT_H
 #include <linux/input.h>
+#include <asm/ioctl.h>
 #include "xlat/evdev_abs.h"
 #include "xlat/evdev_autorepeat.h"
 #include "xlat/evdev_ff_status.h"
diff -rup a/mknod.c b/mknod.c
--- a/mknod.c	2015-07-28 11:38:22.962503148 +0200
+++ b/mknod.c	2015-07-28 11:39:09.953616349 +0200
@@ -1,6 +1,7 @@
 #include "defs.h"
 
 #include <fcntl.h>
+#include <sys/stat.h>
 
 #ifdef MAJOR_IN_SYSMACROS
 # include <sys/sysmacros.h>
diff -rup a/printmode.c b/printmode.c
--- a/printmode.c	2015-07-28 11:43:32.819555915 +0200
+++ b/printmode.c	2015-07-28 11:42:52.310407674 +0200
@@ -1,6 +1,7 @@
 #include "defs.h"
 
 #include <fcntl.h>
+#include <sys/stat.h>
 
 #include "xlat/modetypes.h"
 
